Added Django Sudo functionality for instructor dashboard and course team page

waheedahmed opened the pull request:
[PLAT-407] (https://openedx.atlassian.net/browse/PLAT-407), [PLAT-394] (https://openedx.atlassian.net/browse/PLAT-394)
Decorated instructor dashboard with sudo_required and added couple of unit tests.
Decorated studio permissions with sudo_required.

Sandbox: http://django-sudo.sandbox.edx.org/

Already reviewed in https://github.com/edx/edx-platform/pull/7404, https://github.com/edx/edx-platform/pull/9180 and https://github.com/edx/edx-platform/pull/9552.

### Steps To Enable Third Party Auth
1. Go to this link and follow the steps [Getting-started-with-edx-third-party] (http://johnmcox.blogspot.com/2014/05/getting-started-with-edx-third-party.html)

waheedahmed commented:
jenkins run python

waheedahmed commented:
@adampalay @cpennington @explorerleslie @cahrens FYI.

waheedahmed commented:
jenkins run lettuce

frrrances commented on cms/static/sass/elements/_modal.scss:
```diff
@@ -70,3 +70,129 @@
 	width: 100%;
 	background: $black;
 }
+
```
Remove this sass when you move it out of a modal.

frrrances commented on lms/templates/sudo/sudo.html:
```diff
@@ -0,0 +1,21 @@
+{% load i18n %}
+
+{% block body %}
+<section style="margin: 0 auto; width: 480px; padding: 50px;">
```
Remove inline styles.

frrrances commented on lms/templates/sudo/sudo.html:
```diff
@@ -0,0 +1,21 @@
+<section style="margin: 0 auto; width: 480px; padding: 50px;">
+    <div class="inner-wrapper">
+        <header>
+          <h2 style="text-align: center;">{% trans "Confirm Your Password to Access the Instructor Dashboard" %}</h2>
```
Remove inline styles.

frrrances commented on lms/templates/sudo/sudo.html:
```diff
@@ -0,0 +1,21 @@
+          <h2 style="text-align: center;">{% trans "Confirm Your Password to Access the Instructor Dashboard" %}</h2>
+        </header>
+        <hr />
+        <div style="margin: 0px auto; width: 218px;">
```
Remove inline styles.

waheedahmed commented:
jenkins run python

waheedahmed commented:
jenkins run bokchoy

waheedahmed commented:
@cptvitamin Final review please?

waheedahmed commented:
@dsjen, @dan-f I added couple of events for django-sudo in this commit https://github.com/edx/edx-platform/commit/ba20815978bb1893a53ad3d2289b7e56b34ba6ca, please review.

Studio was crashing after rebasing with master due to a new model added here https://github.com/edx/edx-platform/commit/f69304373ca8663adeb995a25252e824239a2e84#diff-fe2e2bebef5c5b8dc202c1768d240c6eR572. 
To fix this I added ```provider``` and ```provider.oauth2``` in installed apps of CMS and also added dependency for migration which actually fixed the issue.
This change isn't related to analytics.

@adampalay FYI.

waheedahmed commented:
jenkins run python

waheedahmed commented:
jenkins run lettuce

waheedahmed commented:
jenkins run bokchoy

waheedahmed commented:
jenkins run bokchoy

waheedahmed commented:
jenkins run bokchoy

waheedahmed commented:
jenkins run bokchoy, lettuce, python

waheedahmed commented:
jenkins run lettuce
jenkins run python

waheedahmed commented:
@shnayder Please review this.

FYI: @stroilova, @adampalay

adampalay commented on common/djangoapps/django_sudo_helpers/utils.py:
```diff
@@ -0,0 +1,52 @@
+
+    user_role = "django_admin"
+    if course_key:
+        user_role = get_user_role(user, course_key)
```
this won't capture if a user is `is_staff`

waheedahmed commented:
@adampalay does it looks good now?

waheedahmed commented:
jenkins run bokchoy

waheedahmed commented on common/djangoapps/third_party_auth/settings.py:
```diff
@@ -84,3 +85,15 @@ def apply_settings(django_settings):
+    """
+    from social.strategies.utils import set_current_strategy_getter
+    from social.apps.django_app.utils import load_strategy
+    set_current_strategy_getter(load_strategy)
```
@doctoryes, @cpennington Please take a look at this.

**Background**
```
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 108, in get_response
    response = middleware_method(request)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/student/middleware.py", line 19, in process_request
    user_account = UserStanding.objects.get(user=user.id)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/functional.py", line 225, in inner
    self._setup()
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/functional.py", line 365, in _setup
    self._wrapped = self._setupfunc()
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/middleware.py", line 22, in <lambda>
    request.user = SimpleLazyObject(lambda: get_user(request))
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/middleware.py", line 10, in get_user
    request._cached_user = auth.get_user(request)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/__init__.py", line 174, in get_user
    user = backend.get_user(user_id)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/social/backends/base.py", line 188, in get_user
    return strategy.get_user(user_id)
AttributeError: 'NoneType' object has no attribute 'get_user'
```
I was getting this issue after django1.8 upgrade, when I logged into LMS using social auth and go to studio then this error was occurring. 

And I think reason of this issue is here in [python-social-auth] (https://github.com/omab/python-social-auth/blob/master/social/apps/django_app/__init__.py#L16-L21).

So to fix this I added the above function, what do you guys think about this?

cpennington replied:
My question would be, why does python-social-auth restrict that workaround to django 1.7 and below? It seems like they think it should work in django 1.8 without that code. Can you try and track down how python-social-auth is expecting to work in django 1.8?

waheedahmed replied:
Thanks @cpennington, you're right.

cahrens commented:
Given the age of this PR, I assume it can be closed.
